src/placeos-driver/driver_model.cr

```diff
@@ -22,16 +22,32 @@ struct PlaceOS::Driver::DriverModel
     include JSON::Serializable
 
     def initialize(
-      @functions = {} of String => Hash(String, Array(JSON::Any)),
+      @interface : Hash(String, Hash(String, JSON::Any))? = nil,
```

Comment:
Can this type be tightened to reflect the JSON schema objects as the values?

Reply:
it can't as this is used to parse the functions that other drivers are publishing
and at this point in the compilation process the macro expansions are not complete so we don't have all the information available to us

Reply:
Was more referencing defining a `JSON::Schema` type to reflect that structure, rather than any specific usage of this.